patch/cisco-acpi-spi-nor.patch

```diff
@@ -0,0 +1,46 @@
+From d7cdcd35762d7c0957fd356b65456bf3cb952b73 Mon Sep 17 00:00:00 2001
+From: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+Date: Tue, 20 Jul 2021 12:22:47 -0700
+Subject: [PATCH] mtd: m25p80 spi driver update to support ACPI table match
+
+Current m25p80 spi nor driver does not support ACPI table match.
+Configuration done through ACPI tables is not recognized by driver.
+Cisco 8000 platform configures NOR flash partition information
+through ACPI table which is not recognized by m25p80.
+
+Added support in m25p80 to perform ACPI table match.
+
+Signed-off-by: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+---
+ drivers/mtd/devices/m25p80.c | 10 ++++++++++
+ 1 file changed, 10 insertions(+)
+
+diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
+index c4a1d04b8..9b7bd47c5 100644
+--- a/drivers/mtd/devices/m25p80.c
++++ b/drivers/mtd/devices/m25p80.c
+@@ -319,11 +319,21 @@ static const struct of_device_id m25p_of_table[] = {
+ };
+ MODULE_DEVICE_TABLE(of, m25p_of_table);
+ 
++#ifdef CONFIG_ACPI
++static const struct acpi_device_id m25p_acpi_table[] = {
++	{ "JEDEC,SPI-NOR", 0 },
++	{ "ACPI0000", 0 },
```

Comment:
is this a standard name? will this have any conflict with existing acpi device?

Reply:
@msiddare to evaluate whether ACPI0000 is still needed or not.

Reply:
This is still unanswered.

Reply:
Hi Paul,

"ACPI0000" is required for the detecting the MTD partition. 
I have experimented removing "ACPI000" to check if MTD partitions are recognized by driver during probe and they are not. 
This patch is only for 4.19.152 kernel, will port changes to 5.10 according to new file.

Thanks
Madhava